### PR TITLE
Statefulset Permanent Volume claim templates

### PR DESCRIPTION
**Pull request by benoittoulme**
Please ensure your pull request adheres to the following guidelines:
- [ x] explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make

* enabling  extra podLabels and extra annotations to satisfy our company requirements related to service metadata. We think this is a common need that will benefit others.
* Use Volume Claim Templates in the sonarqube-sts template. This modernizes the way Volume claims are handled.

- [x ] Document your Changes in the `CHANGELOG.md` file of the respected chart as well as the `Chart.yaml`

### CODE REVIEW

File: charts/sonarqube/templates/sonarqube-sts.yaml [hunk 1]
```diff
@@ -22,6 +31,32 @@ spec:
...
+        requests:
+          storage: {{ .Values.persistence.size | quote }}
+    {{- if .Values.persistence.storageClass }}
+    {{- if (eq "-" .Values.persistence.storageClass) }}
+      storageClassName: ""
+    {{- else }}
+      storageClassName: "{{ .Values.persistence.storageClass }}"
+    {{- end }}
+    {{- end }}
+  {{- end }}
```
Comment (benoittoulme) [inline charts/sonarqube/templates/sonarqube-sts.yaml:59]: The changes to this file are similar to those done in this PR: https://github.com/SonarSource/helm-chart-sonarqube/pull/96 for the sonarqube-dce chart.

**Comment by benoittoulme** [comment]
@jCOTINEAU Sorry if I overstep here, but I think the github account of the reviewer for this PR does not exist anymore? At least it seems like so. So maybe this PR is in a review limbo. 
Apologies if that is not the case.

**Comment by jCOTINEAU** [comment]
Hello @benoittoulme,
The user is still active in the company, nonetheless, we learned the hard way that if you open too many PR too fast to change codeowner license file, github may consider you are a Spam Bot and block your account.

Situation will be back to normal soon.

**Comment by carminevassallo** [comment]
Hi @benoittoulme,

Thanks for your contribution and sorry for the delay in handling this review. We actually build our own roadmap and try to keep with reviewing external contributions at best effort basis. I would like to review and possibly merge your contribution by end of next week.
 
Could you please rebase the changes made on master? For example, there was a fix on the CI that should be incorporated in order to test your proposed changes on our build environment...

**Comment by carminevassallo** [comment]
Hi @benoittoulme,

Thanks again for your contribution. Following our external PR evaluation process, I need to close this PR because I don't see any further reaction from you. 

Feel free to answer my points and reopen it. Those answers are essential to assess its usefulness.

**Pull request closed without merging**